src/main/java/org/motechproject/mots/service/UserService.java

```diff
@@ -20,42 +26,105 @@
 @Service
 public class UserService {
 
+  public static final String INCHARGE_USER_ROLE = "Incharge";
+
   @Autowired
   private UserRepository userRepository;
 
   @Autowired
   private RoleRepository roleRepository;
 
+  @Autowired
+  AuthenticationHelper authenticationHelper;
+
   private UserMapper userMapper = UserMapper.INSTANCE;
 
+  public User getUserByUserName(String userName) {
+    return userRepository.findOneByUsername(userName).orElseThrow(() ->
+        new EntityNotFoundException("User with username: {0} not found", userName));
+  }
+
   @PreAuthorize(RoleNames.HAS_MANAGE_USERS_ROLE)
   public Iterable<User> getUsers() {
     return userRepository.findAll();
   }
 
   /**
-   * Finds users matching all of the provided parameters.
-   * If there are no parameters, return all users.
+   * Finds users matching all of the provided parameters. If there are no parameters, return all
+   * users.
    */
-  @PreAuthorize(RoleNames.HAS_MANAGE_USERS_ROLE)
+  @PreAuthorize(RoleNames.HAS_MANAGE_USERS_OR_MANAGE_INCHARGE_USERS_ROLE)
   public Page<User> searchUsers(String username, String email, String name, String role,
       Pageable pageable) throws IllegalArgumentException {
-    return userRepository.search(username, email, name, role, pageable);
+    if (authenticationHelper.getCurrentUser().hasPermission(UserPermission.MANAGE_USERS)) {
+      return userRepository.search(username, email, name, role, pageable);
+    }
+
+    final Optional<UserRole> incharge = inchargeRole();
```

Comment:
when I was writing to rename this, I mean the variable name, not the method, rename the variable "incharge" to inchargeRole, the name of the method was good, so change it back or change it to "getInchargeRole"